client/order/epochqueue_test.go

```diff
@@ -207,4 +287,86 @@ func TestEpochQueue(t *testing.T) {
 		t.Fatalf("expected commitment checksum %x, got %x",
 			expectedCmtChecksum, cmtChecksum)
 	}
+
+	// Ensure match proof fails when there is a preimage mismatch.
+	preimages = []order.Preimage{n1Pimg}
+	_, _, err = eq.GenerateMatchProof(preimages, nil)
+	if err == nil {
+		t.Fatalf("[GenerateMatchProof] expected a preimage/orders mismatch")
+	}
+
+	preimages = []order.Preimage{n1Pimg, n2Pimg, n3Pimg}
+	_, _, err = eq.GenerateMatchProof(preimages, nil)
+	if err == nil {
+		t.Fatalf("[GenerateMatchProof] expected no order match " +
+			"for preimate error")
+	}
 }
+
+func benchmarkGenerateMatchProofAlt(c int, b *testing.B) {
+	for n := 0; n < b.N; n++ {
+		mid := "mkt"
+		preimages := make([]order.Preimage, c)
+		commitments := make([]order.Commitment, c)
+		eq := NewEpochQueue()
+
+		for i := 0; i < c; i++ {
+			pimg := order.Preimage{}
+			copy(pimg[:], encode.RandomBytes(32))
+			commitment := makeCommitment(pimg)
+			oid := order.OrderID{}
+			copy(oid[:], encode.RandomBytes(32))
+			note := makeEpochOrderNote(mid, msgjson.BuyOrderNum, oid, uint64(i+1), uint64((i+1)*2), commitment)
+			err := eq.Enqueue(note)
+			if err != nil {
+				b.Fatalf("[Enqueue]: unexpected error: %v", err)
+			}
+
+			preimages[i] = pimg
+			commitments[i] = commitment
+		}
+
+		_, _, err := eq.GenerateMatchProofAlt(preimages, nil)
```

Comment:
But this benchmark still uses the Enqueue with the unneeded operations.  A correct comparison needs two different EpochQueue types.
Also, the test data generation should not be part of the bench mark.  e.g. https://github.com/chappjc/dcrdex/commit/1bb44d76c801981bc4b7bc03bf8e78b55fcffca9#diff-e17d0f6ebad116a19390b508c97ef4f5R65-R96

Reply:
I added `EnqueueAlt` for that but forgot use it in the benchmark.